.travis.yml

```diff
@@ -68,8 +14,77 @@ branches:
 cache:
   pip: true
 
-after_success:
-  # only submit code coverage for the newest python version
-  - if [[ $TRAVIS_PYTHON_VERSION == "3.6" ]]; then
-      coveralls;
-    fi
+jobs:
+  include:
+    - stage: test
+      python: 3.5
+      install:
+        - pip install tox
+        - pip install tox-travis
+      script:
+        - tox
+    - &supported_python_build
+      stage: test
+      python: 3.6
+      install:
+        - pip install tox
+        - pip install tox-travis
+        - pip install coveralls
+      script:
+        - tox
+      after_success: coveralls
+    - &docker_build
+      stage: build
+      python: 3.6
+      services:
+        - docker
+      env:
+        - DOCKER_IMAGE=stodevx/cs251-toolkit
+      script:
+        - script/docker-build $DOCKER_IMAGE:$TRAVIS_COMMIT
```

Comment:
Since this is just building, and we're not using this artifact anywhere else, could we merge this into the previous stage so it can run in parallel with the tests?